### DISCUSSION TITLE
Gap support

### DISCUSSION THREAD

**Discussion by mihkeleidast** [TanStack/virtual#375] [Ideas]
Using the gap property on flex and grid items is quite common these days. It does not seem like this library supports using them, though - any way to make it possible?

Example: https://codesandbox.io/s/interesting-einstein-juxpi0?file=/src/index.css

I'd expect the flex/grid layouts and respective gap properties to be respected.

**Comment by tannerlinsley** [comment]
Virtualization is an absolutely positioned layout. You can hack around it with nested elements with padding. But we Will not be supporting gap directly

**Comment by mihkeleidast** [comment]
Sure, but in order to support gap, it would need to be taken into account in the max width/height and position calculations. Why not add props to the hook to specify row/column gap? Is there a more specific reason why this couldn't be done?

I am aware of the possible workarounds, but that's what they are - hacky workarounds. I was looking forward to trying out this headless approach to see if it improves anything (we currently use react-virtuoso, which does support gap, but is not hook-based), so quite sad if this is not even considered.
